az-core/src/main/java/azkaban/utils/JSONUtils.java

```diff
@@ -44,50 +44,50 @@
   private JSONUtils() {
   }
 
-  public static String toJSON(final Object obj) {
+  public static String toJSON(Object obj) {
```

Comment:
It looks like your save action is doing reverse thing. It should keep final for parameters as mentioned in https://github.com/azkaban/azkaban/blob/master/CONTRIBUTING.md. Can you please fix it? It will also reduce number of lines to review in this PR. It will be easy to review the PR after that.